gpii/node_modules/processReporter/src/ProcessReporter.js

```diff
@@ -0,0 +1,205 @@
+/**
+ * GPII Process Reporter
+ *
+ * Copyright 2015 OCAD University
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii");
+
+fluid.require("kettle", require);
+fluid.require("./processesBridge.js", require);
+
+fluid.defaults("gpii.processReporter", {
+    gradeNames: "fluid.component"
+});
+
+fluid.defaults("gpii.processReporter.base", {
+    gradeNames: ["kettle.app", "gpii.processReporter"],
+    requestHandlers: {
+        processGet: {
+            route: "/processes",
+            type: "gpii.processReporter.allSolutions.get",
+            method: "get"
+        },
+        processGetSingle: {
+            route: "/processes/:solutionID",
+            type: "gpii.processReporter.solution.get",
+            method: "get"
+        }
+    },
```

Comment:
Perhaps an odd question, but what's the motivation for having this component available through REST-endpoints? I think this will always live alongside a flowmanager process. I know that we've historically built most of the components with REST APIs, but I think we've also slowly started to move away from that practice, unless we know it's needed by one of our deployment setups... So this question is because I dont really know when we're making our components restful, and when we're not... And which category this should fall under.. Perhaps @amb26 has some input here